ta4j-core/src/test/java/org/ta4j/core/criteria/EnterAndHoldCriterionTest.java

```diff
@@ -47,116 +46,116 @@ public EnterAndHoldCriterionTest(NumFactory numFactory) {
     }
 
     @Test
-    public void calculateOnlyWithGainPositions() {
-        var series = new MockBarSeriesBuilder().withNumFactory(numFactory)
-                .withData(100, 105, 110, 100, 95, 105)
-                .build();
-        TradingRecord tradingRecord = new BaseTradingRecord(Trade.buyAt(0, series), Trade.sellAt(2, series),
-                Trade.buyAt(3, series), Trade.sellAt(5, series));
+    public void calculateWithOnePosition() {
+        var series = new MockBarSeriesBuilder().withNumFactory(numFactory).withData(100, 105).build();
+        var position = new Position(Trade.buyAt(0, series), Trade.sellAt(1, series));
 
         // buy and hold of ReturnCriterion
-        AnalysisCriterion buyAndHoldReturn = getCriterion(new ReturnCriterion());
-        assertNumEquals(1.05, buyAndHoldReturn.calculate(series, tradingRecord));
+        var buyAndHoldReturn = getCriterion(new ReturnCriterion());
+        assertNumEquals(105d / 100, buyAndHoldReturn.calculate(series, position));
 
         // sell and hold of ReturnCriterion
-        AnalysisCriterion sellAndHoldReturn = getCriterion(TradeType.SELL, new ReturnCriterion());
-        assertNumEquals(0.95, sellAndHoldReturn.calculate(series, tradingRecord));
+        var sellAndHoldReturn = getCriterion(TradeType.SELL, new ReturnCriterion());
+        assertNumEquals(0.95, sellAndHoldReturn.calculate(series, position));
 
-        // buy and hold of ProfitLossPercentageCriterion
-        AnalysisCriterion buyAndHoldPnlPercentage = getCriterion(new ProfitLossPercentageCriterion());
-        assertNumEquals(5, buyAndHoldPnlPercentage.calculate(series, tradingRecord));
+        // buy and hold of PnlPercentageCriterion
+        var buyAndHoldPnlPercentage = getCriterion(new ProfitLossPercentageCriterion());
+        assertNumEquals(5, buyAndHoldPnlPercentage.calculate(series, position));
 
-        // sell and hold of ProfitLossPercentageCriterion
-        AnalysisCriterion sellAndHoldPnlPercentage = getCriterion(TradeType.SELL, new ProfitLossPercentageCriterion());
-        assertNumEquals(-5, sellAndHoldPnlPercentage.calculate(series, tradingRecord));
+        // sell and hold of PnlPercentageCriterion
+        var sellAndHoldPnlPercentage = getCriterion(TradeType.SELL, new ProfitLossPercentageCriterion());
+        assertNumEquals(-5, sellAndHoldPnlPercentage.calculate(series, position));
     }
 
```

Comment:
There should be additional unit tests that exercise the new `amount` functionality. Among them should be the `null` case and the explicitly set `amount` case.